qiskit/pulse/library/waveform.py

```diff
@@ -93,6 +94,13 @@ def _clip(self, samples: np.ndarray, epsilon: float = 1e-7) -> np.ndarray:
 
         return samples
 
+    def assign_parameters(self,
+                          value_dict: Dict[ParameterExpression,
+                                           Union[ParameterExpression, int, float, complex]]
```

Comment:
I think we should consider defining a type for this such as `PulseValue` or something like that. Type hints are getting long.

Reply:
I agree. I avoided doing this because it has to be done in every file -- but I think parameterexpression.py is the right place for it